### PR TITLE
Clean up astar and locations config

### PR DESCRIPTION
**Pull request by thommann**
- astar cleanup extraction into classes + inheritance
- owner config has max and min locations (instead of positions string)

### CODE REVIEW

File: Bidding/BiddingAStar.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+class BiddingAStar(AStar):
+    def __init__(self,
+                 environment: "Environment",
+                 max_iter: int = 100_000,
+                 g_sum: float = 0.5,
+                 height_adjust: Optional[str] = None):
+        super().__init__(environment, max_iter, g_sum, height_adjust)
+        self.node_class = BiddingNode
+
+    def is_valid_for_allocation(self, position: "Coordinate4D", agent: "BiddingABAgent"):
```
Comment (joelbarmettlerUZH) [inline Bidding/BiddingAStar.py]: Indicate return value for this function

File: Bidding/BiddingAStar.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        if self.environment.is_blocked(position, agent.near_radius, agent.speed):
+            return False, set()
+        agents = self.environment.intersect(position, agent.near_radius, agent.speed)
+        res = set()
```
Comment (joelbarmettlerUZH) [inline Bidding/BiddingAStar.py]: Choose a better variable name than "res"

File: Bidding/BiddingAllocator.py [hunk 3]
```diff
@@ -25,7 +25,11 @@ def compatible_owner():
     def __init__(self):
         super().__init__()
 
-    def allocate_for_agents(self, agents: List["BiddingABAgent"], env: "Environment", tick: int) -> list["PathReallocation"]:
+    def allocate_for_agents(self,
+                            agents: List["BiddingABAgent"],
+                            env: "Environment",
+                            tick: int) -> List["PathReallocation"]:
+        astar = BiddingAStar(env)
         res = []
```
Comment (joelbarmettlerUZH) [inline Bidding/BiddingAllocator.py]: Choose a better variable name than "res"

File: FCFS/FCFSAllocator.py [hunk 4]
```diff
@@ -28,6 +28,7 @@ def allocate_for_agents(self,
                             agents: List[Agent],
                             env: Environment,
                             tick: int) -> List[SpaceReallocation | PathReallocation]:
+        astar = AStar(env)
         res = []
```
Comment (joelbarmettlerUZH) [inline FCFS/FCFSAllocator.py]: Choose a better variable name than "res"

File: Simulator/AStar/AStar.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                 max_iter: int = 100_000,
+                 g_sum: float = 0.2,
+                 height_adjust: bool = True):
+        self.node_class = Node
+        self.environment: "Environment" = environment
+        self.max_iter: int = max_iter
+        self.g_sum: float = g_sum
+        self.height_adjust: bool = height_adjust
+
+    def valid_start(self, start: "Coordinate4D", agent: "PathAgent", in_air: bool):
```
Comment (joelbarmettlerUZH) [inline Simulator/AStar/AStar.py]: Indicate return type of function

File: Simulator/AStar/AStar.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    def distance15(start: "Coordinate4D", end: "Coordinate4D"):
+        return math.pow(abs(start.x - end.x) ** 2 + abs(start.y - end.y) ** 2 + abs(start.z - end.z) ** 2, 0.333)
+
+    @staticmethod
+    def distance2(start: "Coordinate4D", end: "Coordinate4D"):
+        return math.pow((start.x - end.x) ** 2 + (start.y - end.y) ** 2 + (start.z - end.z) ** 2, 0.5)
+
+    @staticmethod
+    def distance3(start: "Coordinate4D", end: "Coordinate4D"):
+        return math.pow((start.x - end.x) ** 4 + (start.y - end.y) ** 4 + (start.z - end.z) ** 4, 1 / 4)
```
Comment (joelbarmettlerUZH) [inline Simulator/AStar/AStar.py:169]: Shouldn't distance3 be qubed?

Comment (thommann) [reply Simulator/AStar/AStar.py:169]: I do not know

**Review by joelbarmettlerUZH — APPROVED** [review]

**Pull request merged**